api/src/main/java/io/minio/PutObjectOptions.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage,
+ * (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+import io.minio.errors.InvalidArgumentException;
+import java.util.Map;
+
+public class PutObjectOptions {
+  // allowed maximum object size is 5TiB.
+  public static final long MAX_OBJECT_SIZE = 5L * 1024 * 1024 * 1024 * 1024;
+  // allowed minimum part size is 5MiB in multipart upload.
+  public static final int MIN_MULTIPART_SIZE = 5 * 1024 * 1024;
+  // allowed minimum part size is 5GiB in multipart upload.
+  public static final long MAX_PART_SIZE = 5L * 1024 * 1024 * 1024;
+  public static final int MAX_MULTIPART_COUNT = 10000;
+
+  private long objectSize = -1;
+  private long partSize = -1;
+  private int partCount = -1;
+  private String contentType = "application/octet-stream";
+  private Map<String, String> headers = null;
+  private ServerSideEncryption sse = null;
```

Comment:
can we also add `legal-hold, `retention` and `tags` as part of this?

Reply:
Ashish will add those.

Reply:
then should that be done before we make a release?

Reply:
Is this PR release blocker?

Reply:
We need to have this pr in first and then i will add legal hold/ retention and tags

Reply:
Release can be done once retention pr is merged on top of this pr